test/test_coretasks.py

```diff
@@ -4,128 +4,111 @@
 
 import pytest
 
-from sopel import coretasks
+from sopel import config
 from sopel.module import VOICE, HALFOP, OP, ADMIN, OWNER
 from sopel.tools import Identifier
-from sopel.test_tools import MockSopel, MockSopelWrapper
-from sopel.trigger import PreTrigger, Trigger
+from sopel.tests import rawlist
 
 
 @pytest.fixture
-def sopel():
-    bot = MockSopel("Sopel")
-    return bot
+def tmpconfig(tmpdir):
+    conf_file = tmpdir.join('conf.ini')
+    conf_file.write("\n".join([
+        "[core]",
+        "owner = Uowner",
+        "nick = TestBot",
+        "enable = coretasks"
+        ""
+    ]))
+    return config.Config(conf_file.strpath)
 
 
-def test_bot_mixed_modes(sopel):
-    """
-    Ensure mixed modes like +vha are tracked correctly.
-    Sopel 6.6.6 and older would assign all modes to all users. #1575
-    """
+@pytest.fixture
+def mockbot(tmpconfig, botfactory):
+    return botfactory.preloaded(tmpconfig)
 
-    # RPL_NAMREPLY to create Users and (zeroed) privs
-    for user in set("Unothing Uvoice Uhalfop Uop Uadmin Uowner".split(" ")):
-        pretrigger = PreTrigger(
-            "Foo", ":test.example.com 353 Foo = #test :Foo %s" % user
-        )
-        trigger = Trigger(sopel.config, pretrigger, None)
-        coretasks.handle_names(MockSopelWrapper(sopel, trigger), trigger)
 
-    pretrigger = PreTrigger("Foo", "MODE #test +qvhao Uowner Uvoice Uhalfop Uadmin Uop")
-    trigger = Trigger(sopel.config, pretrigger, None)
-    coretasks.track_modes(MockSopelWrapper(sopel, trigger), trigger)
+def test_bot_mixed_modes(mockbot, ircfactory):
+    """Ensure mixed modes like ``+vha`` are tracked correctly.
 
-    assert sopel.channels["#test"].privileges[Identifier("Unothing")] == 0
-    assert sopel.channels["#test"].privileges[Identifier("Uvoice")] == VOICE
-    assert sopel.channels["#test"].privileges[Identifier("Uhalfop")] == HALFOP
-    assert sopel.channels["#test"].privileges[Identifier("Uop")] == OP
-    assert sopel.channels["#test"].privileges[Identifier("Uadmin")] == ADMIN
-    assert sopel.channels["#test"].privileges[Identifier("Uowner")] == OWNER
+    Sopel 6.6.6 and older would assign all modes to all users.
 
+    .. seealso::
 
-def test_bot_mixed_mode_removal(sopel):
-    """
-    Ensure mixed mode types like -h+a are handled
-    Sopel 6.6.6 and older did not handle this correctly. #1575
+        GitHub issue #1575.
```

Comment:
Want to link to it?

Reply:
I lazily assumed that any readers will be able to go on github and figure out their way to the issue. On the other hand, who knows if this code will still be on github later? I might add the link in the end.